pkg/helpers/crd.go

```diff
@@ -26,14 +42,27 @@ func CreateOrUpdateGitRepo(obj metav1.Object, gvk schema.GroupVersionKind, templ
 		return fmt.Errorf("the tenant name is empty")
 	}
 
+	if template.DeletionPolicy == "" {
+		template.DeletionPolicy = GetDeletionPolicy()
+	}
+
 	if template.RepoType == synv1alpha1.DefaultRepoType {
 		template.RepoType = synv1alpha1.AutoRepoType
 	}
 
+	for file, content := range template.TemplateFiles {
+		if content == manager.DeletionMagicString {
+			delete(template.TemplateFiles, file)
+		}
+	}
+
 	repo := &synv1alpha1.GitRepo{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      obj.GetName(),
 			Namespace: obj.GetNamespace(),
+			Annotations: map[string]string{
+				DeleteProtectionAnnotation: obj.GetAnnotations()[DeleteProtectionAnnotation],
```

Comment:
If the annotation is not set on `obj` this will return an empty string which in turn is interpreted as `true`

Reply:
Actually there's already `AddDeletionProtection()` that handles this. Changing it to use that.